internal/pkg/addon/addons_test.go

```diff
@@ -51,12 +51,72 @@ func TestAddons_Template(t *testing.T) {
 					ws:     ws,
 				}
 			},
-			wantedErr: &ErrAddonsDirNotExist{
+			wantedErr: &ErrAddonsNotFound{
 				WlName:    testJobName,
 				ParentErr: testErr,
 			},
 		},
-		"print correct error message for ErrAddonsDirNotExist": {
+		"return ErrAddonsNotFound if addons directory is empty in a service": {
+			mockAddons: func(ctrl *gomock.Controller) *Addons {
+				ws := mocks.NewMockworkspaceReader(ctrl)
+				ws.EXPECT().ReadAddonsDir(testSvcName).
+					Return([]string{}, nil)
+				return &Addons{
+					wlName: testSvcName,
+					ws:     ws,
+				}
+			},
+			wantedErr: &ErrAddonsNotFound{
+				WlName:    testSvcName,
+				ParentErr: nil,
+			},
+		},
+		"return ErrAddonsNotFound if addons directory is empty in a job": {
```

Comment:
nit: looks like we can remove this test, it's the same one as above where just the name is changed

Reply:
Thanks @efekarakus 🙏🏾. I was a bit confused about this too. I followed the pattern in `return ErrAddonsNotFound if addons doesn't exist in a service` and `return ErrAddonsNotFound if addons doesn't exist in a job` tests nevertheless. I thought the original author may have included this to detect that workload name is correctly reflected in error (i.e. not a fixed string 😄). I will remove them now.